src/Akeneo/Tool/Bundle/MessengerBundle/Command/ProcessMessageCommand.php

```
@@ -0,0 +1,82 @@
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class ProcessMessageCommand extends Command
+{
+    protected static $defaultName = 'akeneo:process-message';
+    /** @var array<string, TraceableMessageHandlerInterface> */
+    private array $handlers = [];
+
+    public function __construct(
+        private readonly SerializerInterface $serializer,
+        private readonly LoggerInterface $logger
+    ) {
+        parent::__construct();
+    }
+
+    public function registerHandler(TraceableMessageHandlerInterface $handler, string $consumerName)
+    {
+        $this->handlers[$consumerName] = $handler;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->addArgument('consumer_name', InputArgument::REQUIRED, 'consumer name')
+            ->addArgument('message_class', InputArgument::REQUIRED, 'class of the message')
+            ->addArgument('message', InputArgument::REQUIRED, 'message in json')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $message = $this->serializer->deserialize(
+            $input->getArgument('message'),
+            $input->getArgument('message_class'),
+            'json'
+        );
+
+        $consumerName = $input->getArgument('consumer_name');
+        $handler = $this->handlers[$consumerName] ?? null;
+
+        if (null === $handler) {
+            throw new \Exception(sprintf('No handler found for the "%s" consumer', $consumerName));
+        }
+
+        try {
+            ($handler)($message);
+        } catch (\Throwable $t) {
+            $context = ['trace' => $t->getTraceAsString()];
+            if ($message instanceof CorrelationAwareInterface) {
+                $context['correlation_id'] = $message->getCorrelationId();
+            }
+
+            $this->logger->error(sprintf('An error occurred: %s', $t->getMessage()), $context);
+
+            throw $t;
```

Do we really need to throw the error ?
There is a log just before and as the command is launched from a message consumer, maybe that just `return Command:FAILURE` is enough